include/drivers/uart.h

```diff
@@ -723,12 +748,14 @@ static inline int z_impl_uart_config_get(struct device *dev,
 static inline int uart_fifo_fill(struct device *dev, const uint8_t *tx_data,
 				 int size)
 {
+#ifdef CONFIG_UART_INTERRUPT_DRIVEN
 	const struct uart_driver_api *api =
 		(const struct uart_driver_api *)dev->driver_api;
 
 	if (api->fifo_fill) {
 		return api->fifo_fill(dev, tx_data, size);
 	}
+#endif
 
 	return 0;
```

Comment:
Should this return -ENOTSUP?  I'm not sure.

Reply:
this function does not return error code. It returns `Number of bytes sent.` I was wondering if it should assert here but lack of define is more or less the same as lack of function pointer in api struct and that returned 0.